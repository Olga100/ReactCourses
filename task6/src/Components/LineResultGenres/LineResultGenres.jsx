import React from 'react';
import PropTypes from 'prop-types';

import './LineResultGenres.scss';

const LineResultGenres = () => (<div className="lineResultGenresContainer">lineResultGenresContainer</div>)

/*
LineResultMovieList.propTypes = {
    moviesCount: PropTypes.number,
};
*/

export default LineResultGenres;
