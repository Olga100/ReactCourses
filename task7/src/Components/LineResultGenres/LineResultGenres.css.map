{
"version": 3,
"mappings": "AAAA,0BAA2B;EACzB,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;EACnB,MAAM,EAAE,IAAI;EACZ,YAAY,EAAE,IAAI;EAClB,WAAW,EAAE,GAAG;EAChB,gBAAgB,EAAE,OAAO",
"sources": ["LineResultGenres.scss"],
"names": [],
"file": "LineResultGenres.css"
}