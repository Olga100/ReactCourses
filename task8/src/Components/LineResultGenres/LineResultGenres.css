.lineResultGenresContainer {
  display: flex;
  align-items: center;
  height: 30px;
  padding-left: 50px;
  font-weight: 600;
  background-color: #f5f5f5; }

/*# sourceMappingURL=LineResultGenres.css.map */
