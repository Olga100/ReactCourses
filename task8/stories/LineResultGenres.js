import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from "@storybook/addon-info";

import SearchPage from './../src/Containers/SearchPage/SearchPage';
import {getQuery} from "../src/Reducers/mainReducer";


const movies = ["action", "fantasy"];

const searchPageWith = {
    movies: movies,
    sortBy: "rating",
    searchBy: "Star wars",
    searchText: state.searchText,
    query: getQuery(state)
};



const Red = props => <span style={{ color: "red" }} {...props} />;

const TableComponent = ({ propDefinitions }) => {
    const props = propDefinitions.map(
        ({ property, propType, required, description, defaultValue }) => {
            return (
                <tr key={property}>
                    <td>
                        {property}
                        {required ? <Red>*</Red> : null}
                    </td>
                    <td>{propType.name}</td>
                    <td>{defaultValue}</td>
                    <td>{description}</td>
                </tr>
            );
        },
    );

    return (
        <table>
            <thead>
            <tr>
                <th>name</th>
                <th>type</th>
                <th>default</th>
                <th>description</th>
            </tr>
            </thead>
            <tbody>{props}</tbody>
        </table>
    );
};

storiesOf('SearchPage', module)
    .add('searchPageWith1', () => <SearchPage movies={searchPageWith} />)
    .add('searchPageWith2', withInfo({})(() => <SearchPage movies={searchPageWith} />));
