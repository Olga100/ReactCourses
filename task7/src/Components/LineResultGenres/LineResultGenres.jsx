import React from 'react';
import PropTypes from 'prop-types';

import './LineResultGenres.scss';

const LineResultGenres = ({genres}) => {
    return genres === null
        ? <div className="lineResultGenresContainer">Films by unknown genre</div>
        : <div className="lineResultGenresContainer">Films by {genres.join(', ')} genres</div>
};

LineResultGenres.propTypes = {
    genres: PropTypes.array,
};

export default LineResultGenres;
