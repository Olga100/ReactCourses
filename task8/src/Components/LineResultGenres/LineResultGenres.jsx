// @flow

import React from 'react';


import './LineResultGenres.scss';

type Props = {
    genres: Array<string>
};


const LineResultGenres = ({genres}: Props) => {
    return genres === null
        ? <div className="lineResultGenresContainer">Films by unknown genre</div>
        : <div className="lineResultGenresContainer">Films by {genres.join(', ')} genres</div>
};

LineResultGenres.defaultProps = {
    genres: null
};

export default LineResultGenres;
