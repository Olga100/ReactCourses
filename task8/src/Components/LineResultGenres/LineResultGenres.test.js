import React from 'react';
import renderer from 'react-test-renderer';

import LineResultGenres from './LineResultGenres';


describe ('LineResultGenres', () => {

    it('should match snapshot with data', () => {
        const component = renderer.create(
            <LineResultGenres  genres={["horror", "action"]}  />
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it('should match snapshot without data', () => {
        const component = renderer.create(
            <LineResultGenres  genres={null}/>
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

});