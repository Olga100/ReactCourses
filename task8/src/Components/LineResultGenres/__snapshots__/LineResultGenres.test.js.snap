// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineResultGenres should match snapshot with data 1`] = `
<div
  className="lineResultGenresContainer"
>
  Films by 
  horror, action
   genres
</div>
`;

exports[`LineResultGenres should match snapshot without data 1`] = `
<div
  className="lineResultGenresContainer"
>
  Films by unknown genre
</div>
`;
